lib/modules/manager/composer/utils.ts

```diff
@@ -75,7 +75,22 @@ export function extractConstraints(
 
   // extract php
   if (composerJson.config?.platform?.php) {
-    res.php = composerJson.config.platform.php;
+    const major = api.getMajor(composerJson.config.platform.php);
+    const minor = api.getMinor(composerJson.config.platform.php);
+    const patch = api.getPatch(composerJson.config.platform.php);
+
+    // handle where x is taken by composer to mean lowest patch of the lowest minor (x.0.0)
+    if (minor === null || minor === 0) {
```

Comment:
What if the version is 7.0.5? Is it intentional that you'd use 7.0.0 instead?

Reply:
ah good spot! Ty, I will update to:
`${major}.0.${patch}`